db_stress_tool/db_stress_common.cc

```diff
@@ -70,7 +128,16 @@ int64_t GenerateOneKey(ThreadState* thread, uint64_t iteration) {
       static_cast<double>(iteration) / FLAGS_ops_per_thread;
   const int64_t base_key = static_cast<int64_t>(
       completed_ratio * (FLAGS_max_key - FLAGS_active_width));
-  return base_key + thread->rand.Next() % FLAGS_active_width;
+  int64_t rand_seed = base_key + thread->rand.Next() % FLAGS_active_width;
+  int64_t cur_key = rand_seed;
+  if (FLAGS_hot_key_alpha != 0) {
+    // If set the Zipfian distribution Alpha to non 0, use Zipfian
+    double float_rand =
+        (static_cast<double>(thread->rand.Next() % FLAGS_max_key)) /
+        FLAGS_max_key;
+    cur_key = GetOneHotKeyID(float_rand, FLAGS_max_key);
```

Comment:
Shouldn't we still do the base_key and FLAGS_active_width thing for hot keys?

Reply:
@siying I think we cannot use these. base_key is like a starting point of a key range, from 0 move to (max_key - active_width) based on the completion ratio. Keys are randomly generated in the key range. So overall, the probability of each key being accessed are the same. However, if we use the active_width and do Zipfian in each range, the probability of each key being access are accumulated as the "keyrange window" slides. Statistically, each key is accessed with the same probability. If we do batch by batch (one batch of accesses only happen in a active_width key range), the behavior will be strange.

Reply:
OK then.